src/odemis/acq/stream/_base.py

```diff
@@ -561,6 +636,65 @@ def _getDetectorVA(self, vaname):
                 raise AttributeError(u"Detector has not VA %s" % (vaname,))
             return hwva
 
+    def _linkHwAxes(self):
+        """"
+        Link the axes to the hardware components
+        If local axis Vigilant Attributes's are specified, write the values of the local axis VA
+        to the real hardware
+        """
+
+        if hasattr(self, "_axis_map") and self._axis_map is not None:
+
+            logging.info("Moving linked HW axes")
+
+            moving_axes = []
+            for ax_name in self._axis_map:
+                try:
+                    local_pos_va = self._axis_vas[ax_name]
+                    act = self._axis_map[ax_name][1]
+                    pos = local_pos_va.value
+                    logging.info("Moving actuator %s to position %s.", act.name, pos)
+                    f = act.moveAbs({ax_name: pos})
+                    moving_axes.append(f)
+                    # subscribe to update the axis when the stream plays
+                    updater = functools.partial(self._update_linked_axis, ax_name)
+                    self._axisvaupdaters[ax_name] = updater
+                    local_pos_va.subscribe(updater)
+                except Exception:
+                    logging.exception("Failed to move axis.")
+
+            for f in moving_axes:
+                try:
+                    f.result()
+                except Exception:
+                    logging.exception("Failed to move axis.")
+
+    def _update_linked_axis(self, ax_name, pos):
+        """ Update the value of a linked hardware axis VA
+            when the stream is active
+        """
+        if not self.is_active.value:
+            return
+        if hasattr(self, "_axis_map") and self._axis_map:
+            try:
+                act = self._axis_map[ax_name][1]
+                logging.info("Moving actuator %s to position %s.", act.name, pos)
+                f = act.moveAbs({ax_name: pos})
+                # f.result()
```

Comment:
Please add a comment here why you'd want this here: this would allow the caller (ie, the GUI) to know when the axis is done moving. However, this would require the GUI to be adjusted so that it sets the value in a separate thread.